rust/protocol/src/address.rs

```diff
@@ -1,26 +1,57 @@
 //
-// Copyright 2020 Signal Messenger, LLC.
+// Copyright 2020-2021 Signal Messenger, LLC.
 // SPDX-License-Identifier: AGPL-3.0-only
 //
 
+//! Where to send a [sealed_sender] message to.
+
+// These are used in docstrings.
+#[allow(unused_imports)]
+use crate::sealed_sender::{self, UnidentifiedSenderMessageContent};
+
 use std::fmt;
 
+/// The type used in memory to represent a *device*, i.e. a particular Signal client instance which
+/// represents some user.
+///
+/// Used in [ProtocolAddress].
+pub type DeviceId = u32;
+
+/// The target of an [UnidentifiedSenderMessageContent].
```

Comment:
The core type here is SignalMessage, not UnidentifiedSenderMessageContent, but even then ProtocolAddress is used for purposes beyond that. (For example, it's used when encrypting the new multi-recipient sealed sender messages, even though the USMC of such a message is the same for each recipient.) I think it'd be better to just say that ProtocolAddress represents a unique Signal client instance as a (user ID, device ID) pair.

Reply:
This is super helpful, thanks!